### ISSUE TITLE
board_register_gcloud isn't working

### ISSUE THREAD

**Issue by augustohassel** [rstudio/pins#326]
> sessionInfo()
R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)
Matrix products: default
locale:
[1] LC_COLLATE=Spanish_Argentina.1252  LC_CTYPE=Spanish_Argentina.1252    LC_MONETARY=Spanish_Argentina.1252
[4] LC_NUMERIC=C                       LC_TIME=Spanish_Argentina.1252    
attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     
other attached packages:
 [1] digest_0.6.25   pins_0.4.4      log4r_0.3.2     xlsx_0.6.4.2    xtable_1.8-4    rmarkdown_2.4  
 [7] knitr_1.30      shiny_1.5.0     sodium_1.1      mailR_0.4.1     jsonlite_1.7.1  httr_1.4.2     
[13] RJDBC_0.2-8     rJava_0.9-13    odbc_1.2.3      lubridate_1.7.9 forcats_0.5.0   stringr_1.4.0  
[19] dplyr_1.0.2     purrr_0.3.4     readr_1.4.0     tidyr_1.1.2     tibble_3.0.4    ggplot2_3.3.2  
[25] tidyverse_1.3.0 RMySQL_0.10.20  DBI_1.1.0      
loaded via a namespace (and not attached):
 [1] bit64_4.0.5       R.utils_2.10.1    modelr_0.1.8      assertthat_0.2.1  blob_1.2.1        xlsxjars_0.6.1   
 [7] cellranger_1.1.0  yaml_2.2.1        pillar_1.4.6      backports_1.1.10  glue_1.4.2        promises_1.1.1   
[13] rvest_0.3.6       colorspace_1.4-1  htmltools_0.5.0   httpuv_1.5.4      R.oo_1.24.0       pkgconfig_2.0.3  
[19] broom_0.7.1       haven_2.3.1       scales_1.1.1      later_1.1.0.1     generics_0.0.2    ellipsis_0.3.1   
[25] withr_2.3.0       cli_2.1.0         magrittr_1.5      crayon_1.3.4      readxl_1.3.1      mime_0.9         
[31] evaluate_0.14     R.methodsS3_1.8.1 fs_1.5.0          fansi_0.4.1       xml2_1.3.2        tools_4.0.3      
[37] hms_0.5.3         lifecycle_0.2.0   munsell_0.5.0     reprex_0.3.0      compiler_4.0.3    rlang_0.4.8      
[43] grid_4.0.3        rstudioapi_0.11   rappdirs_0.3.1    gtable_0.3.0      curl_4.3          R6_2.5.0         
[49] fastmap_1.0.1     bit_4.0.4         stringi_1.5.3     Rcpp_1.0.5        vctrs_0.3.4       dbplyr_1.4.4     
[55] tidyselect_1.1.0  xfun_0.18  

As of today, the function `board_register_gcloud()` has stopped working, giving me this error: 

> Error in yaml.load(string, error.label = error.label, ...) : 
  (C:\Users\augus\AppData\Local\Temp\Rtmpo1FB3S\file31b829fc1105) Scanner error: while scanning a simple key at line 70, column 3 could not find expected ':' at line 71, column 1

Nothing has changed on my code, it was working perfectly until yesterday. Today Google had an issue on their services. I don't know if this could be related.

I cannot give a fully reprex because one need to create a bucket on google in order to connect. 

Anyone with the same issue?

**Comment by javierluraschi** [comment]
Just tried this, worked for me:

<img width="446" alt="Screen Shot 2020-12-14 at 12 46 17 PM" src="https://user-images.githubusercontent.com/3478847/102133737-6b8e1200-3e0a-11eb-906b-71e020ba1533.png">

Did you perhaps updated the `gcloud` SDK? I wonder if the API changed there.

**Comment by augustohassel** [comment]
I'm not using the gcloud SDK, I'm passing directly the token. This code has been running in production for about two months without any changes. 

I cannot directly connect to the bucket, one step privous of using `pin_get()`

I'm suspisciuos that it doesn't have to do with the token because I get the same error with and without referencing it.

> board_register_gcloud(bucket = "test_bucket", token = "exampletoken")
Error in yaml.load(string, error.label = error.label, ...) : 
  (C:\Users\augus\AppData\Local\Temp\Rtmpo1FB3S\file31b86e59191d) Scanner error: while scanning a simple key at line 70, column 3 could not find expected ':' at line 71, column 1

> board_register_gcloud(bucket = "test_bucket")
Error in yaml.load(string, error.label = error.label, ...) : 
  (C:\Users\augus\AppData\Local\Temp\Rtmpo1FB3S\file31b87b9a6e5c) Scanner error: while scanning a simple key at line 70, column 3 could not find expected ':' at line 71, column 1

Today started this behaviour.. =/

**Comment by augustohassel** [comment]
I've also tried deleting what's inside `board_cache_path()`, just in case, but nothing has change

**Comment by augustohassel** [comment]
@javierluraschi I think I've found something. 
I've created another bucket (an empty one), then try to connect with board_register_gcloud() and worked all fine.
On the other hand, the previous bucket is still not working, but I discovered that the temp file that's been created and that has the metadata is beeing cuted out at the lenght 1656, exaclty were the error is showing. I'm attaching the temp file on a txt format.

[file61741ec53ec8.txt](https://github.com/rstudio/pins/files/5692345/file61741ec53ec8.txt)

Hope this helps.

**Comment by javierluraschi** [comment]
@augustohassel which version of `pins` do you have installed? You can get the version with `packageVersion("pins")`

I'd try upgrading `pins` from CRAN if you are not in `0.4.4`. Older version of pins did not supported concurrent updates but that should be fixed.

Can you explain a bit more of your environment? Is this happening from your own computer? Or are you perhaps running R in a server?

Could it be possible is that somehow you are running out of disk space?

Could you try manually fixing the `data.txt` in Google Cloud and retry? Could you try experimenting with the size? If you remove a few entries from `data.txt`, does the problem get fixed? Is a particular pin causing this? Can you reproduce with other pins?

Would love to get to the bottom of this but we will need some help from you. Thanks for reporting this and helping us out!

**Comment by augustohassel** [comment]
@javierluraschi I've made a few tries and this is what I've found: If I correct the data.txt manually, then it will start to work properly again and even if I try to create more files inside the bucket (more than the lenght that I tought it was the limit) nothing wrongs happen. This leads to me that's not likely to be a problem of the pins package.

As I reported the problem we were having other issues regarding Google services, so maybe at that point something happend that corrupted the data.txt file and with that file incomplete this problem arise. 

I'm running a process in production saving raw results during the day every 15 minutes, so this package has been a lot of help. Congrats for it and thanks for the rush response.

PS: I'm using the latest version of the package: 0.4.4
